bns-test/test/name_import.test.ts

```diff
@@ -0,0 +1,172 @@
+      expect(receipt.success).eq(false);
+      expect(receipt.error).include('Aborted: 1011');
+    // });
+
+    // it("Bob trying to import 'alpha.blockstack' should succeed", async () => {
+      receipt = await bns.nameImport(cases[0].namespace, "alpha", cases[0].zonefile, { sender: bob })
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('Returned: true');
+    // });
+
+    // it("Resolving an imported name should succeed if the namespace is not ready", async () => {
+      receipt = await bns.getNameZonefile(
+        cases[0].namespace, 
+        "alpha", { sender: cases[0].nameOwner });
+
+      expect(receipt.result).include('0x30303030');
+      expect(receipt.success).eq(true);
+    // });
+
+    // it("Bob trying to import 'beta.blockstack' should fail after the launch of the domain", async () => {
+      receipt = await bns.namespaceReady(cases[0].namespace, { sender: bob });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+      await bns.mineBlocks(1);  
+
+      receipt = await bns.nameImport(cases[0].namespace, "beta", cases[0].zonefile, { sender: bob })
+      expect(receipt.success).eq(false);
+      expect(receipt.error).include('1014');
+    // });
+
+    // it("Resolving an imported name should succeed if the namespace is ready", async () => {
+      receipt = await bns.getNameZonefile(
+        cases[0].namespace, 
+        "alpha", { sender: bob });
+      expect(receipt.result).include('0x30303030');
+      expect(receipt.success).eq(true);
+    // });
+
+    // it("Charlie trying to register 'alpha.blockstack' should fail", async () => {
+      receipt = await bns.namePreorder(
+        cases[0].namespace,
+        "alpha",
+        cases[0].salt, 
+        160000, { sender: charlie });
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('u22');
+
+      receipt = await bns.nameRegister(
+        cases[0].namespace, 
+        "alpha", 
+        cases[0].salt, 
+        cases[0].zonefile, { sender: charlie });
+      expect(receipt.error).include('2004');
+      expect(receipt.success).eq(false);
+    // });
+
+    // it("Charlie trying to renew 'alpha.blockstack' should fail", async () => {
+      receipt = await bns.nameRenewal(cases[0].namespace, "alpha", 160000, charlie, cases[0].zonefile, { sender: charlie })
+      expect(receipt.success).eq(false);
+      expect(receipt.error).include('2006');
+    // });
+  
+    // it("Bob trying to renew 'alpha.blockstack' should succeed", async () => {
+      receipt = await bns.nameRenewal(cases[0].namespace, "alpha", 160000, charlie, "6666", { sender: bob })
+      expect(receipt.success).eq(true);
+      expect(receipt.result).include('true');
+    // });
+
+    // it("Resolving an imported name should fail after expiration", async () => {
+      await bns.mineBlocks(100);
```

Comment:
Can this be expanded to test the corner case?  `alpha.blockstack` should last 10 blocks after the namespace is launched, right?  Can you verify that the name still resolves at the 10th block, but not the 11th block?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/4f68dbfebb7c29bb5028d615dd826ff488e9de85